x-pack/plugins/security/public/management/roles/edit_role/privileges/es/index_privilege_form.tsx

```diff
@@ -345,6 +360,24 @@ export class IndexPrivilegeForm extends Component<Props, State> {
     );
   };
 
+  private editorDidMount = (editor: monaco.editor.IStandaloneCodeEditor) => {
+    /**
+     * Resize the editor based on the contents of the editor itself.
+     * Adapted from https://github.com/microsoft/monaco-editor/issues/794#issuecomment-688959283
+     */
+
+    const minHeight = 100;
+    const maxHeight = 1000;
+
+    const updateHeight = () => {
+      const contentHeight = Math.min(maxHeight, Math.max(minHeight, editor.getContentHeight()));
+      this.setState({ documentQueryEditorHeight: `${contentHeight}px` });
+    };
+
+    editor.onDidContentSizeChange(updateHeight);
+    updateHeight();
+  };
```

Comment:
This is great, maybe we could update the component in core so that other pages get the same functionality (ability to specify `maxHeight` prop and automatically resizing the field accordingly)

Reply:
Good idea, will do!

Reply:
I started exploring this, and it would actually end up being a decent amount of work, so I'd prefer to defer this until we have a need to share this code. We would need to share state between the base shared editor and its EUI form field wrapper, and those two aren't setup to share that sort of state today